src/libs/OptimisticReportNames.ts

```diff
@@ -368,10 +409,16 @@ function updateOptimisticReportNamesFromUpdates(updates: OnyxUpdate[], context:
     }
 
     Log.info('[OptimisticReportNames] Processing completed', false, {
-        additionalUpdatesCount: additionalUpdates.length,
+        additionalOptimisticUpdatesCount: additionalOptimisticUpdates.length,
+        additionalSuccessUpdatesCount: additionalSuccessUpdates.length,
+        additionalFailureUpdatesCount: additionalFailureUpdates.length,
     });
 
-    return updates.concat(additionalUpdates);
+    return {
+        optimisticData: optimisticData.concat(additionalOptimisticUpdates),
+        successData: successData ? successData.concat(additionalSuccessUpdates) : undefined,
```

Comment:
Shouldn't we always return the additional updates even if the original data is empty?

```suggestion
        successData: successData ? successData.concat(additionalSuccessUpdates) : additionalSuccessUpdates,
```